src/core/thread/router_table.hpp

```diff
@@ -265,6 +265,16 @@ class RouterTable : public InstanceLocator
      */
     Router *GetRouter(const Mac::ExtAddress &aExtAddress);
 
+    /**
+     * This method returns the router table index for a given `Router` instance.
+     *
+     * @param[in]  aRouter  A reference to a `Router`
+     *
+     * @returns The index corresponding to @p aRouter.
+     *
+     */
+    size_t GetRouterIndex(const Router &aRouter) const { return static_cast<uint16_t>(&aRouter - mRouters); }
```

Comment:
I have a suggestion here:
How about instead of this, we add a method like 
```
    /**
     * This method indicates whether a given neighbor is in the router table.
     *
     * @note This method checks whether the `aNeighbor` instance itself is an entry from the router table.
     *
     * @param[in] aNeighbor   A reference to a neighbor.
     *
     * @returns TRUE if the neighbor is present in the router table, FALSE otherwise.
     *
     */
    bool ContainsNeighbor(const Neighbor &aNeighbor) const;
```

Two reason for this:
- I think it will help simplify the code where we use this to check.
- The fact that `RouterTable` uses an array to store router entries is a private info in the `RouterTable`. Publicly, the `RouterTable` provides methods to search/find a entry and an `Iterator` to go through the entries (which does now show what the internal data structure is). The whole notion of router table "index" (which is only applicable when we know we have an array) is not part of public APIs of `RouterTable`.  Using `ContainsNeighbor` would be more generic (it allows us in future to consider changing internal data structure used to store routers from array).

Reply:
Agree. Will revise. I have actually done it but reverted back to have less code changes. 

Do you think the method name should be `Contains` rather than `ContainsNeighbor` because the argument type already says it's a Neighbor.

Reply:
Thanks. 
Yes `Contains` sounds good to me.

Reply:
revised accordingly